### DISCUSSION TITLE
Autocomplete inside template, js and css in small components on intellij products

### DISCUSSION THREAD

**Discussion by daniboygg** [django-components/django-components#548] [Ideas]
I saw this issue https://github.com/EmilStenstrom/django-components/issues/183 and using the 2024 version of IntelliJ products language injections seems to work pretty well (I didn't test it too much though). 

It is ok add to the documentation to achieve this in IntelliJ products? Maybe add documentation for getting this for Neovim as well, or is broadening too much this topic? 

Here is how you can achieve the same stated in the documentation https://github.com/EmilStenstrom/django-components?tab=readme-ov-file#using-single-file-about but with IntelliJ products with language injections https://www.jetbrains.com/help/pycharm/using-language-injections.html

```python
# In a file called [project root]/components/calendar.py
from django_components import component

@component.register("calendar")
class Calendar(component.Component):
    def get_context_data(self, date):
        return {
            "date": date,
        }

    # language=html
    template= """
        <div class="calendar-component">Today's date is <span>{{ date }}</span></div>
    """

    # language=css
    css = """
        .calendar-component { width: 200px; background: pink; }
        .calendar-component span { font-weight: bold; }
    """

    # language=js
    js = """
        (function(){
            if (document.querySelector(".calendar-component")) {
                document.querySelector(".calendar-component").onclick = function(){ alert("Clicked calendar!"); };
            }
        })()
    """
```

**Comment by EmilStenstrom** [comment]
@daniboygg I think adding instructions for this is something that would fit well in the documentation (both VSCode, IntelliJ and Neovim). Would you mind adding that as a PR?

**Reply by daniboygg** [reply]
Yes, I would make the PR as soon as I have a moment.

Talking about neovim, I'm a novice, so I don't have the confidence to write documentation, but I can try to open an issue in the repo for someone else to do.

A couple of questions: 

1. It would be safe to assume since this is a django project, the product would be PyCharm, right?
2. Reading this issue https://github.com/EmilStenstrom/django-components/issues/401, I thought that documentation would change its structure, is this coming soon? Maybe it would be better to wait until is made to fit the documentation in the right place? Otherwise, I would make the PR right away.

**Reply by EmilStenstrom** [reply]
@daniboygg Then let's skip neovim for now. I think most Django developers use VSCode for now, with maybe PyCharm as a second place. I'm sure IntelliJ is in the list too somewhere. 

About the move to a new strucutre, it's not coming soon I'm afraid, so let's act as if it didn't exist.

**Comment by JuroOravec** [comment]
Closing as this has been implemented in [v0.85](https://github.com/EmilStenstrom/django-components/releases/tag/0.85)
